lib/decoder/index.js

```diff
@@ -12,6 +12,16 @@ function base64decode(str) {
   }
 }
 
+function decodeURIComponentCompletely(str, step) {
```

Comment:
I think this is fine to do, but from my reading I wouldn't expect things to ever be encoded more than twice. It seems like the some encodings are preserved in the logged value of the URL path, and CF has a list of characters that it encodes for all fields in the log line. So really the _only_ thing I would ever expect to get double encoded is `%`.

But I don't think there's any harm in being this aggressive, unless there comes a day where we need to maintain some semantic distinction between `/` or some other URL-meaningful character.